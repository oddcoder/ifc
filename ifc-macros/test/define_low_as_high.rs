use ifc_macros::ifc_block;

fn main() {
    ifc_block!{
        let x : usize = 5;
        #[IFC(Low)]
        let _y: usize  = x;
    };
}