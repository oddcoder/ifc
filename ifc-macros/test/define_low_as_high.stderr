error: Cannot assign high expression to low variable
 --> $DIR/define_low_as_high.rs:7:9
  |
7 |         let _y: usize  = x;
  |         ^^^^^^^^^^^^^^^^^^^
  |
help: Low Variable
 --> $DIR/define_low_as_high.rs:7:26
  |
7 |         let _y: usize  = x;
  |                          ^
help: High Expression
 --> $DIR/define_low_as_high.rs:4:5
  |
4 | /     ifc_block!{
5 | |         let x : usize = 5;
6 | |         #[IFC(Low)]
7 | |         let _y: usize  = x;
8 | |     };
  | |______^
  = note: this error originates in the macro `ifc_block` (in Nightly builds, run with -Z macro-backtrace for more info)
